deepmedic/frontEnd/trainSession.py

```diff
@@ -19,148 +19,186 @@
 
 from deepmedic.routines.training import do_training
 
+from deepmedic.logging.tensorboard_logger import TensorboardLogger
+
 import tensorflow as tf
 
+
 class TrainSession(Session):
-    
+
     def __init__(self, cfg):
         self._out_folder_models = None
         self._out_folder_preds = None
         self._out_folder_fms = None
-        self._params = None # Compiled from cfg. Required for run()
+        self._params = None  # Compiled from cfg. Required for run()
         Session.__init__(self, cfg)
-        
+
     def _make_sess_name(self):
-        sess_name = TrainSessionParameters.getSessionName(  self._cfg[self._cfg.SESSION_NAME] )
+        sess_name = TrainSessionParameters.getSessionName(self._cfg[self._cfg.SESSION_NAME])
         return sess_name
-    
+
     def make_output_folders(self):
-        self._main_out_folder_abs = getAbsPathEvenIfRelativeIsGiven( self._cfg[self._cfg.FOLDER_OUTP],
-                                                                     self.get_abs_path_to_cfg() )
+        self._main_out_folder_abs = getAbsPathEvenIfRelativeIsGiven(self._cfg[self._cfg.FOLDER_OUTP],
+                                                                    self.get_abs_path_to_cfg())
         [self._log_folder_abs,
          self._out_folder_models,
          self._out_folder_preds,
-         self._out_folder_fms] = makeFoldersNeededForTrainingSession( self._main_out_folder_abs, self._sess_name )
-    
-    
-    def _print_vars_in_collection(self, collection, coll_name="no_name"):
+         self._out_folder_fms] = makeFoldersNeededForTrainingSession(self._main_out_folder_abs, self._sess_name)
+
+    def create_tensorboard_loggers(self, logger_types, tf_graph, create_log=False):
+        tensorboard_loggers = {}
+
+        self._log.print3("----------- Creating Tensorboard Loggers -----------")
+
+        for logger_type in logger_types:
+            if create_log:
+                tensorboard_loggers[logger_type] = TensorboardLogger(os.path.join(self._main_out_folder_abs,
```

Comment:
Create the foldeer in:
makeFoldersNeededForTrainingSession (L46 above)

Reply:
Let's discuss how to best approach this.